static/components.ts

```diff
@@ -980,3 +938,275 @@ export function getDeviceViewWith(
+ */
+function validateComponentState(componentName: string, state: any): boolean {
+    // Basic validation - state must be an object
+    if (typeof state !== 'object' || state === null) {
+        return false;
+    }
+
+    switch (componentName) {
+        case COMPILER_COMPONENT_NAME:
+            // Compiler states can have various combinations of properties
+            return (
+                (state.lang && state.source !== undefined) ||
+                (state.source !== undefined && state.compiler) ||
+                (state.lang && state.tree !== undefined)
+            );
+
+        case EXECUTOR_COMPONENT_NAME:
+            // Executor states require compilation panel booleans
+            return typeof state.compilationPanelShown === 'boolean' && typeof state.compilerOutShown === 'boolean';
+
+        case EDITOR_COMPONENT_NAME:
+            // Editor states are flexible but must have valid properties
+            return true;
+
+        case TREE_COMPONENT_NAME:
+            // Tree states are flexible but must have valid properties
+            return true;
+
+        case OUTPUT_COMPONENT_NAME:
+            // Output state needs specific numeric properties
+            return (
+                typeof state.tree === 'number' && typeof state.compiler === 'number' && typeof state.editor === 'number'
+            );
+
+        case TOOL_COMPONENT_NAME:
+            // Tool state needs specific properties
+            return (
+                typeof state.tree === 'number' &&
+                typeof state.toolId === 'string' &&
+                typeof state.id === 'number' &&
+                typeof state.editorid === 'number'
+            );
+
+        // View components have diverse state requirements but must be valid objects
+        case TOOL_INPUT_VIEW_COMPONENT_NAME:
+        case DIFF_VIEW_COMPONENT_NAME:
+        case OPT_VIEW_COMPONENT_NAME:
+        case STACK_USAGE_VIEW_COMPONENT_NAME:
+        case FLAGS_VIEW_COMPONENT_NAME:
+        case PP_VIEW_COMPONENT_NAME:
+        case AST_VIEW_COMPONENT_NAME:
+        case GCC_DUMP_VIEW_COMPONENT_NAME:
+        case CFG_VIEW_COMPONENT_NAME:
+        case CONFORMANCE_VIEW_COMPONENT_NAME:
+        case IR_VIEW_COMPONENT_NAME:
+        case CLANGIR_VIEW_COMPONENT_NAME:
+        case OPT_PIPELINE_VIEW_COMPONENT_NAME:
+        case LLVM_OPT_PIPELINE_VIEW_COMPONENT_NAME:
+        case RUST_MIR_VIEW_COMPONENT_NAME:
+        case HASKELL_CORE_VIEW_COMPONENT_NAME:
+        case HASKELL_STG_VIEW_COMPONENT_NAME:
+        case HASKELL_CMM_VIEW_COMPONENT_NAME:
+        case GNAT_DEBUG_TREE_VIEW_COMPONENT_NAME:
+        case GNAT_DEBUG_VIEW_COMPONENT_NAME:
+        case RUST_MACRO_EXP_VIEW_COMPONENT_NAME:
+        case RUST_HIR_VIEW_COMPONENT_NAME:
+        case DEVICE_VIEW_COMPONENT_NAME:
+            return true;
+
+        default:
+            // Unknown component name - this should not happen with proper typing
+            return false;
```

Comment:
Suggest adding a Sentry report or something here so we know it's happening

Reply:
Done